src/main/test/java/com/google/sps/TimeRangeTest.java

```diff
@@ -0,0 +1,188 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.time.Instant; 
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public final class TimeRangeTest {
+  @Test
+  public void equality() {
+    Assert.assertEquals(TimeRange(), TimeRange());
+
+    Assert.assertNotEquals(
+        TimeRange.fromStartDuration(100, 50), TimeRange.fromStartDuration(100, 51));
+
+    Assert.assertEquals(
+        TimeRange.fromStartDuration(100, 50), TimeRange.fromStartEnd(100, 150, false));
+
+    Assert.assertNotEquals(
+        TimeRange.fromStartDuration(100, 50), TimeRange.fromStartEnd(100, 150, true));
+
+    Assert.assertEquals(
+        TimeRange.fromStartDuration(100, 51), TimeRange.fromStartEnd(100, 150, true));
+
+    Assert.assertEquals(
+        TimeRange.fromStartEnd(100, 151, false), TimeRange.fromStartEnd(100, 150, true));
+  }
+
+  @Test
+  public void containsPoint() {
+    // Range 100 (inclusive) to 150 (exclusive).
+    TimeRange range = TimeRange.fromStartDuration(100, 50);
+    Assert.assertFalse(range.contains(50));
+    Assert.assertTrue(range.contains(100));
+    Assert.assertFalse(range.contains(150));
+    Assert.assertFalse(range.contains(200));
+  }
+
+  @Test
+  public void containsRange() {
```

Comment:
These containsRange() and overlaps() are really easy to read. Nice!
They are duplicates, however. Remove one?